test/node-rpc-test.js

```diff
@@ -222,6 +224,96 @@ describe('RPC', function() {
       connectionsCnt = await nclient.execute('getconnectioncount', []);
       assert.strictEqual(connectionsCnt, 0);
     });
+
+    it('should rpc getnodeaddresses', async () => {
+      const newHosts = [
+        {
+          'time': 1655305701,
+          'services': 1033,
+          'host': '102.91.5.101',
+          'port': 38333
+        },
+        {
+          'time': 1655958090,
+          'services': 1033,
+          'host': '197.149.96.171',
+          'port': 38333
+        },
+        {
+          'time': 1655834706,
+          'services': 1033,
+          'host': '190.104.161.74',
+          'port': 38331
+        },
+        {
+          'time': 1655621666,
+          'services': 1033,
+          'host': '151.34.40.226',
+          'port': 38333
+        },
+        {
+          'time': 1655965074,
+          'services': 67109901,
+          'host': '178.128.80.131',
+          'port': 3333
+        },
+        {
+          'time': 1656003055,
+          'services': 1033,
+          'host': '31.14.40.18',
+          'port': 38333
+        },
+        {
+          'time': 1654268543,
+          'services': 1033,
+          'host': '102.89.34.71',
+          'port': 38333
+        },
+        {
+          'time': 1655009945,
+          'services': 1033,
+          'host': '183.90.36.72',
+          'port': 38333
+        },
+        {
+          'time': 1655109959,
+          'services': 1033,
+          'host': '151.46.58.162',
+          'port': 38333
+        },
+        {
+          'time': 1653921720,
+          'services': 1033,
+          'host': '5.24.225.133',
+          'port': 38333
+        }
+      ];
+
+      let addr = NetAddress.fromJSON(newHosts[0]);
+      let isHostAdded = node.pool.hosts.add(addr);
+
+      assert.strictEqual(isHostAdded, true);
+
+      let hosts = await nclient.execute('getnodeaddresses');
+
+      assert.strictEqual(addr.host, hosts[0].host);
+      assert.strictEqual(addr.port, hosts[0].port);
+      assert.strictEqual(addr.services, hosts[0].services);
+      assert.strictEqual(addr.time, hosts[0].time);
+
+      // count=7
+      const newHostsCount = newHosts.length;
+
+      for(let i = 1; i < newHostsCount; i++) {
+        addr = NetAddress.fromJSON(newHosts[i]);
+        isHostAdded = node.pool.hosts.add(addr);
+        assert.strictEqual(isHostAdded, true);
+      }
+
+      hosts = await nclient.execute('getnodeaddresses', [7]);
```

Comment:
call this twice, once with no parameters to make sure we get one back. then continue with the `[7]` and make sure we get 7 back.